### PR TITLE
Add jupyterhub grafana dashboard via overlays

### PR DESCRIPTION
**Pull request by hemajv**
Based on the discussion in #87, adding the example Jupyterhub Grafana monitoring dashboard to be included via overlays.

### CODE REVIEW

File: odh/base/monitoring/overrides/grafana-operator/overlays/dashboards/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: opf-monitoring
```
Comment (HumairAK) [inline odh/base/monitoring/overrides/grafana-operator/overlays/dashboards/kustomization.yaml:4]: I don't think we need this namespace field here.

**Comment by HumairAK** [comment]
this lgtm, @tumido  has more experience with overlays for kfdefs so I'd have him take a gander as well.

File: odh/base/monitoring/overrides/grafana-operator/overlays/dashboards/kustomization.yaml (hunk 1 above)
Comment (4n4nd) [reply odh/base/monitoring/overrides/grafana-operator/overlays/dashboards/kustomization.yaml:4]: do we need a kustomization if we are doing overlays?

Comment (hemajv) [reply odh/base/monitoring/overrides/grafana-operator/overlays/dashboards/kustomization.yaml]: fixed

File: odh/base/monitoring/overrides/grafana-operator/overlays/dashboards/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
```
Comment (tumido) [inline odh/base/monitoring/overrides/grafana-operator/overlays/dashboards/kustomization.yaml:4]: What is interesting to notice here @HumairAK @hemajv you don't even need the `base` in the resources in here, despite it being an overlay. The way how kfctl folds and combines overlays and writes a new `kustomization.yaml`  for you from scratch is very magical. :slightly_smiling_face:

**Review by tumido — APPROVED** [review]
Using this `kfdef` as a reproducer:
```yaml
apiVersion: kfdef.apps.kubeflow.org/v1
kind: KfDef
metadata:
  name: opendatahub
spec:
  applications:
    - kustomizeConfig:
        repoRef:
          name: manifests
          path: grafana/grafana
      name: grafana-operator
    - kustomizeConfig:
        overlays:
          - dashboards
        repoRef:
          name: hemajv
          path: odh/base/monitoring/overrides/grafana-operator
      name: grafana-operator
  repos:
    - name: manifests
      uri: "https://github.com/opendatahub-io/odh-manifests/tarball/v0.9.0"
    - name: hemajv
      uri: "https://github.com/hemajv/odh/tarball/add-grafana-dashboards"
  version: v0.9.0
```

When built with kfctl all the resources are present

**Pull request merged**